rio_tiler/io/xarray.py

```diff
@@ -0,0 +1,362 @@
+"""rio_tiler.io.xarray: Xarray Reader."""
+
+import warnings
+from typing import Any, Dict, List, Optional
+
+import attr
+import xarray
+from morecantile import Tile, TileMatrixSet
+from rasterio.crs import CRS
+from rasterio.enums import Resampling
+from rasterio.features import is_valid_geom
+from rasterio.rio.overview import get_maximum_overview_level
+from rasterio.transform import from_bounds, rowcol
+from rasterio.warp import calculate_default_transform
+from rasterio.warp import transform as transform_coords
+
+from rio_tiler.constants import WEB_MERCATOR_TMS, WGS84_CRS
+from rio_tiler.errors import PointOutsideBounds, RioTilerError, TileOutsideBounds
+from rio_tiler.io.base import BaseReader
+from rio_tiler.models import BandStatistics, ImageData, Info, PointData
+from rio_tiler.types import BBox
+
+
+@attr.s
+class XarrayReader(BaseReader):
+    """Xarray Reader.
+
+    Attributes:
+        dataset (rasterio.io.DatasetReader or rasterio.io.DatasetWriter or rasterio.vrt.WarpedVRT, optional): Rasterio dataset.
```

Comment:
It'd be helpful (for me at least) to list out all the restrictions we have on this DataArray:

1. Can be >2d, but requires that the last two dimensions represent longitude / latitude?
2. Do we care what the names of the lat / lon / x / y dimensions are?
3. Must have a CRS as discovered by `.rio.crs`
4. ...

Any others we know of?

Reply:
> Do we care what the names of the lat / lon / x / y dimensions are?

reading https://github.com/corteva/rioxarray/blob/dcb26dd0af034dbffd1952bd61d8f62e16b0ee4d/rioxarray/rioxarray.py#L251-L276 it seems that only `x,y,longitude,latitude` are valid 🤷

Reply:
Or CF coordinates with appropriate standard names.